Content.Server/Glue/GlueSystem.cs

```diff
@@ -1,59 +1,69 @@
 using Content.Shared.IdentityManagement;
 using Content.Shared.Popups;
 using Content.Shared.Item;
-using Content.Shared.Interaction.Components;
 using Content.Shared.Glue;
-using Content.Server.Nutrition.EntitySystems;
 using Content.Shared.Interaction;
+using Content.Server.Chemistry.EntitySystems;
 using Content.Server.Nutrition.Components;
 
-namespace Content.Server.Glue
+namespace Content.Server.Glue;
+
+public sealed class GlueSystem : EntitySystem
 {
-    public sealed class GlueSystem : EntitySystem
+    [Dependency] private readonly SharedAudioSystem _audio = default!;
+    [Dependency] private readonly SharedPopupSystem _popup = default!;
+    [Dependency] private readonly SolutionContainerSystem _solutionContainer = default!;
+
+    public override void Initialize()
     {
-        [Dependency] private readonly SharedAudioSystem _audio = default!;
-        [Dependency] private readonly SharedPopupSystem _popup = default!;
-        [Dependency] private readonly FoodSystem _food = default!;
+        base.Initialize();
 
+        SubscribeLocalEvent<GlueComponent, AfterInteractEvent>(OnInteract);
+    }
 
-        public override void Initialize()
-        {
-            base.Initialize();
+    // When glue bottle is used on item it will apply the glued and unremoveable components.
+    private void OnInteract(EntityUid uid, GlueComponent component, AfterInteractEvent args)
+    {
+        if (args.Handled)
+            return;
 
-            SubscribeLocalEvent<GlueComponent, AfterInteractEvent>(OnInteract);
-        }
+        if (!args.CanReach || args.Target is not { Valid: true } target)
+            return;
 
-        // When glue bottle is used on item it will apply the glued and unremoveable components.
-        private void OnInteract(EntityUid uid, GlueComponent component, AfterInteractEvent args)
+        if (TryComp<DrinkComponent>(uid, out var drink) && !drink.Opened)
         {
-            if (args.Handled)
-                return;
-
-            if (!args.CanReach || args.Target is not { Valid: true } target)
-                return;
+            return;
+        }
 
-            if (HasComp<GluedComponent>(target))
-            {
-                _popup.PopupEntity(Loc.GetString("glue-failure", ("target", Identity.Entity(target, EntityManager))), args.User,
-                args.User, PopupType.Medium);
-                return;
-            }
+        if (TryGlue(uid, component, target))
+        {
+            args.Handled = true;
+            _audio.PlayPvs(component.Squeeze, uid);
+            _popup.PopupEntity(Loc.GetString("glue-success", ("target", Identity.Entity(target, EntityManager))), args.User, args.User, PopupType.Medium);
+        }
+        else
+        {
+            _popup.PopupEntity(Loc.GetString("glue-failure", ("target", Identity.Entity(target, EntityManager))), args.User, args.User, PopupType.Medium);
```

Comment:
im pretty sure `THE` fluent function already does identity stuff, so that is innecesary here